katdal/datasources.py

```diff
@@ -635,19 +635,34 @@ def from_url(cls, url, chunk_store='auto', upgrade_flags=True, **kwargs):
             telstate = katsdptelstate.TelescopeState(katsdptelstate.memory.MemoryBackend())
             try:
                 telstate.load_from_file(url_parts.path)
-            except OSError as err:
-                raise DataSourceNotFound(str(err))
+            except OSError as e:
+                raise DataSourceNotFound(str(e))
         elif url_parts.scheme == 'redis':
             # Redis server
             try:
                 telstate = katsdptelstate.TelescopeState(url_parts.netloc, db)
             except katsdptelstate.ConnectionError as e:
                 raise DataSourceNotFound(str(e))
+        elif url_parts.scheme in {'http', 'https'}:
+            # Treat URL prefix as an S3 object store (with auth info in kwargs)
+            store_path = os.path.dirname(os.path.dirname(url_parts.path))
```

Comment:
Will this work on Windows, where os.path uses `\` as separator but the URL path will use `/`?

One option (that also avoids the need to unparse I think) is
```python
store_url = urllib.parse.urljoin(url, '..')
```

Reply:
I was worrying about the exact same thing but could not find a suitable "url.path.join" in `urllib`. I therefore like your suggestion a lot - will investigate.

Reply:
Fixed as suggested.